internal/dev_server/model/project.go

```diff
@@ -48,6 +48,34 @@ func CreateProject(ctx context.Context, projectKey, sourceEnvironmentKey string,
 	return project, nil
 }
 
+func SyncProject(ctx context.Context, projectKey string) (Project, error) {
+	store := StoreFromContext(ctx)
+	project, err := store.GetDevProject(ctx, projectKey)
+	if err != nil {
+		return Project{}, err
+	}
+	sdkAdapter := adapters.GetSdk(ctx)
+	apiAdapter := adapters.GetApi(ctx)
+	sdkKey, err := apiAdapter.GetSdkKey(ctx, projectKey, project.SourceEnvironmentKey)
+	if err != nil {
+		return Project{}, err
+	}
+	sdkFlags, err := sdkAdapter.GetAllFlagsState(ctx, project.Context, sdkKey)
+	if err != nil {
+		return Project{}, err
+	}
+	project.FlagState = FromAllFlags(sdkFlags)
+	project.LastSyncTime = time.Now()
```

Comment:
Maybe we can extract this into a method on `*Project` that can be reused by the `CreateProject`